src/core/lib/security/transport/security_handshaker.c

```diff
@@ -150,25 +152,33 @@ static void on_peer_checked(grpc_exec_ctx *exec_ctx, void *arg,
     security_handshake_failed_locked(exec_ctx, h, GRPC_ERROR_REF(error));
     goto done;
   }
-  // Get frame protector.
+  // Create frame protector.
   tsi_frame_protector *protector;
-  tsi_result result =
-      tsi_handshaker_create_frame_protector(h->handshaker, NULL, &protector);
+  tsi_result result = tsi_handshaker_result_create_frame_protector(
+      h->handshaker_result, NULL, &protector);
   if (result != TSI_OK) {
     error = grpc_set_tsi_error_result(
         GRPC_ERROR_CREATE_FROM_STATIC_STRING("Frame protector creation failed"),
         result);
     security_handshake_failed_locked(exec_ctx, h, error);
     goto done;
   }
-  // Success.
+  // Get unused bytes.
+  unsigned char *unused_bytes = NULL;
+  size_t unused_bytes_size = 0;
+  result = tsi_handshaker_result_get_unused_bytes(
+      h->handshaker_result, &unused_bytes, &unused_bytes_size);
+  if (unused_bytes_size > 0) {
+    grpc_slice slice =
+        grpc_slice_from_copied_buffer((char *)unused_bytes, unused_bytes_size);
+    grpc_slice_buffer_add(&h->left_overs, slice);
+  }
   // Create secure endpoint.
   h->args->endpoint = grpc_secure_endpoint_create(
       protector, h->args->endpoint, h->left_overs.slices, h->left_overs.count);
```

Comment:
The way that we're now constructing `left_overs`, there will only ever be a single slice in it.  Given that, there's probably no need for a `grpc_slice_buffer` in the first place.  Instead, we can simply pass a pointer to the slice we populated via `grpc_slice_from_copied_buffer()` above and set the length to 1.  We just need to unref the slice right after passing it to `grpc_secure_endpoint_create()`.